server/consumer.go

```diff
@@ -1503,21 +1528,30 @@ func stopAndClearTimer(tp **time.Timer) {
 
 // Stop will shutdown  the consumer for the associated stream.
 func (o *Consumer) Stop() error {
-	return o.stop(false, true)
+	return o.stop(false, true, false)
 }
 
-// Delete will delete the consumer for the associated stream.
+func (o *Consumer) deleteWithoutAdvisory() error {
+	return o.stop(true, true, false)
+}
+
+// Delete will delete the consumer for the associated stream and send advisories.
 func (o *Consumer) Delete() error {
-	return o.stop(true, true)
+	return o.stop(true, true, true)
 }
 
-func (o *Consumer) stop(dflag, doSignal bool) error {
+func (o *Consumer) stop(dflag, doSignal, advisory bool) error {
 	o.mu.Lock()
 	mset := o.mset
 	if mset == nil {
 		o.mu.Unlock()
 		return nil
 	}
+
+	if dflag && advisory {
```

Comment:
Looking at code, do we need advisory flag or just send if dflag true?

Reply:
yeah, see the calls `deleteWithoutAdvisory` where we delete a consumer if right after adding it we discover it should have been gone, so there you dont want to notify deleting one that effectively didnt even finish creating